pkg/config/gateway/kube/inferenceextension.go

```diff
@@ -18,6 +18,7 @@ package kube
 // This data is stored in the `Extra` field of a `config.Config` for a VirtualService.
 // The map is keyed by the `HTTPRoute.Name`.
 type InferencePoolRouteRuleConfig struct {
-	FQDN string
-	Port string
+	FQDN             string
+	Port             string
+	FailureModeAllow bool
```

Comment:
Can we make this an enum. I particularly dislike booleans in interfaces because they tend to get deprecated very easily. For example: https://www.envoyproxy.io/docs/envoy/latest/api-v3/extensions/wasm/v3/wasm.proto#envoy-v3-api-field-extensions-wasm-v3-pluginconfig-fail-open. `fail_open` was replaced by `failure_policy`.

Reply:
sigh, this seems to reflect an upstream API that is already merged... nevermind

Reply:
OH, I thought this was a proto....

Reply:
no, just later it is converted to envoy FailureModeAllow, which is a proto bool upstream btw - https://github.com/envoyproxy/envoy/blob/e439c73e32dfefff0baa4adedfb268c8742a7617/api/envoy/extensions/filters/http/ext_proc/v3/ext_proc.proto#L182